File: src/main/java/io/golem/datasync/domain/RunStatus.java
package io.golem.datasync.domain;

import java.util.EnumSet;
import java.util.Set;

public enum RunStatus {
    SUBMITTING,
    PENDING,
    RUNNING,
    STOPPING,
    SUCCEEDED,
    FAILED,
    CANCELED,
    UNKNOWN;

    private static final Set<RunStatus> TERMINAL = EnumSet.of(SUCCEEDED, FAILED, CANCELED, UNKNOWN);

    public boolean isTerminal() {
        return TERMINAL.contains(this);
    }
}

File: apps/web/src/lib/run-status.ts
import type { RunStatus } from "@/lib/types";

const terminalStatuses: ReadonlySet<RunStatus> = new Set([
  "SUCCEEDED",
  "FAILED",
  "CANCELED",
  "UNKNOWN",
]);

export function isTerminalStatus(status: RunStatus): boolean {
  return terminalStatuses.has(status);
}
